src/test/java/DeadlineTest.java

```diff
@@ -0,0 +1,29 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class DeadlineTest {
+    @Test
+    public void testStringConversion() throws Exception {
+        assertEquals("[D][\u2718] test deadline (by: 1st of January 2019, 06:00PM)",
```

Comment:
Good job on developing unit tests! I appreciate that you developed a wide range of test cases (as compared to what was required). To further improve the logic of the tests, perhaps you could consider catching any exceptions thrown where it is not expected and use the `fail()` method, instead of throwing the exceptions. 

Also, I notice that you are catching and throwing the superclass `Exception` instead of the appropriate `DukeException` subclasses. Perhaps you could consider using the `DukeException` subclasses instead, as this would better capture the logic of "what could go wrong" (as expected by the method) , "what should go wrong" and "what should not go wrong" in your test cases.